src/util-byte.c

```diff
@@ -152,6 +152,9 @@ int ByteExtractUint32(uint32_t *res, int e, uint16_t len, const uint8_t *bytes)
     }
 
     ret = ByteExtract(&i64, e, len, bytes);
+    if (i64 > UINT32_MAX) {
```

Comment:
Not sure why we're doing this. We've ensured that no more than 4 bytes are being extracted ... hence, 32 bits. How can the value exceed `0xffffffff`?

Reply:
Ah yes, this is needed for `ByteExtractStringUint32` but redundant for `ByteExtractUint32`